client/src/main/java/org/apache/celeborn/client/ShuffleClientImpl.java

```diff
@@ -202,21 +201,34 @@ private boolean checkPushBlacklisted(
   private void submitRetryPushData(
       String applicationId,
       int shuffleId,
-      int mapId,
-      int attemptId,
       byte[] body,
       int batchId,
-      PartitionLocation loc,
       RpcResponseCallback wrappedCallback,
       PushState pushState,
-      StatusCode cause,
+      ReviveRequest request,
       int remainReviveTimes) {
+    int mapId = request.mapId;
+    int attemptId = request.attemptId;
+    PartitionLocation loc = request.loc;
+    StatusCode cause = request.cause;
     int partitionId = loc.getId();
-    if (!revive(
-        applicationId, shuffleId, mapId, attemptId, partitionId, loc.getEpoch(), loc, cause)) {
-      wrappedCallback.onFailure(
-          new CelebornIOException(cause + " then revive but " + StatusCode.REVIVE_FAILED));
-    } else if (mapperEnded(shuffleId, mapId)) {
+    long reviveWaitTime =
+        conf.clientRpcRequestPartitionLocationRpcAskTimeout().duration().toMillis() + 1000;
```

Comment:
Here is make sure after wait the revive result must have a result? If so I think we can just wait until it has a result  FAILED, TIMEOUT or SUCCESS. 

+1000 is not safe and we also better add some comment about these logic

Reply:
+1, Will there be too many partitions to be processed, a network timeout may not be enough? May be can limit batch size one request?
BTW, ShuffleClient uses batch to revive by default., is `celeborn.client.shuffle.batchHandleChangePartition.enabled` nessary to be enabled by default in LifeCycleManager?